staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3071,6 +3071,8 @@ type PodSpec struct {
 	// +optional
 	// +patchMergeKey=name
 	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=name
```

Comment:
this addition makes server-side-apply (which is beta prior to 1.22) consistent with client-side-apply and seems reasonable

Reply:
To be honest, we already infer listType and listMapKey if patchMergeKey and patchStrategy is present, so this is mostly a no-op, but yes, it's fine.